src/app/exam/exam-questions/exam-questions.component.ts

```diff
@@ -14,14 +13,15 @@ import {initHighlight, parseMarkdown} from 'src/utils/markdownUtils';
 })
 export class ExamQuestionsComponent implements OnInit, AfterViewInit {
 
-  constructor(private examService: ExamService,
-              private studentService: StudentService,
-              private route: ActivatedRoute,
-              private router: Router,
-              private renderer: Renderer2) {
+  constructor(private http: HttpClient,
```

Comment:
don't perform any HTTP requests in the component, do it in the service.

Reply:
@edisonhello this is not resolved?

should remove the dependency on `HttpClient`

Reply:
Bruh, I forget to push.